src/schematools/events/export.py

```diff
@@ -0,0 +1,143 @@
+"""Exporter module.
+
+This module facilitates the export of events from GOB database tables.
+The main objective of this module is to create full events that can be used
+to test the event-importer that is also part of the `schematools.events` package.
+
+The `schematools` cli interface had an entry to use the code in this module.
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any, Generator
+
+from geoalchemy2.shape import to_shape
+from json_encoder import json
+from sqlalchemy import Table
+from sqlalchemy.engine import Connection
+
+from schematools.events import metadata
+from schematools.events.factories import tables_factory
+from schematools.types import DatasetSchema, DatasetTableSchema
+from schematools.utils import to_snake_case
+
+
+def fetch_complex_fields_info(dataset_table: DatasetTableSchema) -> dict[str, dict]:
+    """Collect info about complex fields (mainly relations)."""
+    complex_fields = {}
+
+    for field in dataset_table.fields:
+        multi = None
+        properties = {}
+        relation_ds = None
+        try:
+            if (nm_relation := field.nm_relation) is not None:
+                multi = True
+                properties = field["items"]["properties"]
+                relation_ds = nm_relation.split(":")[0]
+            if (relation := field.relation) is not None:
+                multi = False
+                properties = field["properties"]
+                relation_ds = relation.split(":")[0]
+        except KeyError:
+            continue
+
+        if multi is not None:
+            complex_fields[to_snake_case(field.name)] = {
+                "multi": multi,
+                "relation_ds": relation_ds,
+                "identifier_names": dataset_table.identifier,
+                "sub_field_names": [to_snake_case(sf) for sf in properties.keys()],
+            }
+    return complex_fields
+
+
+def collect_nm_embed_rows(
+    dataset_id: str,
+    table_id: str,
+    datasets_lookup: dict[str, DatasetSchema],
+    tables: dict[str, dict[str, Table]],
+    complex_fields_info: dict[str, dict],
+    connection: Connection,
+) -> dict[str:Any]:
+    """Fetch row info as list of embeddable objects."""
+    nm_embeds = defaultdict(lambda: defaultdict(list))
+    for field_name, field_info in complex_fields_info.items():
+        if field_info["multi"]:
+            through_table = tables[dataset_id][f"{table_id}_{field_name}"]
+            for row in connection.execute(through_table.select()):
+                row_dict = dict(row)
+                id_value = ".".join(
+                    str(row_dict[f"{table_id}_{idn}"]) for idn in field_info["identifier_names"]
+                )
+
+                stripped_row = {}
+                for sfn in field_info["sub_field_names"]:
+                    stripped_row[sfn] = row_dict[f"{field_name}_{sfn}"]
+                nm_embeds[table_id][id_value].append(stripped_row)
+    return nm_embeds
```

Comment:
```suggestion
    return dict(nm_embeds)
```

is meestal handig, zodat je het defaultdict gedrag kwijt bent voor de aanroeper.

Reply:
Goeie tip. Lijkt me ook handiger in de debugger.